File: src/client/src/services/model/trade.js
import { Direction, TradeStatus, CurrencyPair } from './';

export default class Trade {
  _tradeId:number;
  _traderName:string;
  _currencyPair:CurrencyPair;
  _notional:number;
  _dealtCurrency:string;
  _direction:string;
  _spotRate:number;
  _tradeDate:Date;
  _valueDate:Date;
  _status:TradeStatus;
  _isNew: boolean;
  constructor(
    tradeId:number,
    traderName:string,
    currencyPair:CurrencyPair,
    notional:number,
    dealtCurrency:string,
    direction:Direction,
    spotRate:number,
    tradeDate:Date,
    valueDate:string,
    status:TradeStatus,
    isNew:boolean
  ) {
    this._tradeId = tradeId;
    this._traderName = traderName;
    this._currencyPair = currencyPair;
    this._notional = notional;
    this._dealtCurrency = dealtCurrency;
    this._direction = direction;
    this._spotRate = spotRate;
    this._tradeDate = tradeDate;
    this._valueDate = valueDate;
    this._status = status;
    this._isNew = isNew;
  }

  get tradeId():number {
    return this._tradeId;
  }

  get traderName():string {
    return this._traderName;
  }

  get currencyPair():CurrencyPair {
    return this._currencyPair;
  }

  get notional():number {
    return this._notional;
  }

  get dealtCurrency():string {
    return this._dealtCurrency;
  }

  get direction():Direction {
    return this._direction;
  }

  get spotRate():number {
    return this._spotRate;
  }

  get tradeDate():Date {
    return this._tradeDate;
  }

  get valueDate():Date {
    return this._valueDate;
  }

  get status():TradeStatus {
    return this._status;
  }

  get isNew():boolean {
    return this._isNew;
  }

  set isNew(val: boolean) {
    this._isNew = val;
  }
}

File: src/server/Adaptive.ReactiveTrader.Server.Common/Trade.cs
using System;
using Adaptive.ReactiveTrader.Contract;

namespace Adaptive.ReactiveTrader.Server.Host
{
    public class Trade
    {
        public Trade(
            long tradeId,
            string traderName,
            string currencyPair,
            decimal notional,
            string dealtCurrency,
            DirectionDto direction,
            decimal spotRate,
            DateTimeOffset tradeDate,
            DateTimeOffset valueDate,
            TradeStatusDto status)
        {
            TradeId = tradeId;
            TraderName = traderName;
            CurrencyPair = currencyPair;
            Notional = notional;
            DealtCurrency = dealtCurrency;
            Direction = direction;
            SpotRate = spotRate;
            TradeDate = tradeDate;
            ValueDate = valueDate;
            Status = status;
        }

        public long TradeId { get; }
        public string TraderName { get; }
        public string CurrencyPair { get; }
        public decimal Notional { get; }
        public string DealtCurrency { get; }
        public DirectionDto Direction { get; }
        public decimal SpotRate { get; }
        public DateTimeOffset TradeDate { get; }
        public DateTimeOffset ValueDate { get; }
        public TradeStatusDto Status { get; set; }

        public override string ToString()
        {
            return
                $"TradeId: {TradeId}, TraderName: {TraderName}, CurrencyPair: {CurrencyPair}, Notional: {Notional}, Direction: {Direction}, SpotRate: {SpotRate}, TradeDate: {TradeDate}, ValueDate: {ValueDate}, Status: {Status}, DealtCurrency: {DealtCurrency}";
        }
    }
}
